pyspeckit/spectrum/units.py

```diff
@@ -585,21 +585,24 @@ def umin(self, unit=None):
         else: 
             return self.min()
 
-    def x_to_pix(self, xval, xval_units=None):
+    def x_to_pix(self, xval, xval_units=None, equivalencies=None):
         """
         Given an X coordinate in SpectroscopicAxis' units, return the corresponding pixel number
         """
         if xval_units in pixel_dict:
             return xval
         else:
-            try:
-                if not xval_units:
-                    xval_units = xval.unit
-            except AttributeError:
-                xval_units = u.dimensionless_unscaled
-            xval = xval * u.Unit(xval_units)            
-            nearest_pix = np.argmin(np.abs(self.value-xval.value))
-            # nearest_pix = np.argmin(np.abs(self.value-xval))
+            if xval_units:
+                xval = xval * u.Unit(xval_units)
```

Comment:
It's better performance-wise to use `u.Quantity(xval, xval_units)`, but this should only be done if `not hasattr(xval, 'unit')` (which is a better check than `is not u.Quantity`.

I think this logic could be simplified to:
`xval = u.Quantity(xval, xval_units or self.unit)`